### DISCUSSION TITLE
How to reliably parse the deployment URL from the CLI output?

### DISCUSSION THREAD

**Discussion by borekb** [vercel/vercel#4523] [Help] [answered]
I'd like a script to read the unique deployment URL created by invoking `vercel`, is there a reliable way? In my console, it's all "human-formatted" though on GitHub Actions, it seems that the unique URL is on the last line of the output.

Is that reliable? Is there possibly some flag that sets the output format to something more predictable? (I haven't found anything.)

Thanks.

**Comment by coetry** [comment]
@borekb The two URLs that the CLI prints are after `Inspect: ` and `Preview: `, so you can just parse for URL patterns that follow those two keywords. We don't have any specific commands to adjust the output formatting.

**Reply by borekb** [reply]
Thanks. The "Preview:" is probably replaced by "Production:" when `--prod` is added, I get a unique URL as the last line without any prefix:

```
Now CLI 19.0.1
> NOTE: Deploying to Now 2.0 automatically. More: https://vercel.com/docs/version-detection
- Deploying myorg/project
Inspect: https://vercel.com/myorg/project/kfvoqgv8l [2s]
- Building
- Building
- Completing
Production: https://my-project-domain.now.sh [6s]
https://project-kfvoqgv8l.now.sh
```

**Reply by coetry** [reply]
@borekb Yup! `Production: ` would be available when you add `--prod`. Did you add any extra flags for the output above or did you just run `vc --prod` ?

**Reply by borekb** [reply]
It's basically this:

```
export NOW_PROJECT_ID=fW8vysyskJUPHK2evn91vBJsLv9FKvMvKFmYedz
export NOW_ORG_ID=team_Bt2YTsGM38MrwUOttgQf
now --prod --token "$VERCEL_TOKEN"
```

**Reply by borekb** [reply]
BTW, preview deployments also seem to print the unique URL at the last line:

```
Now CLI 19.0.1
> NOTE: Deploying to Now 2.0 automatically. More: https://vercel.com/docs/version-detection
- Deploying myorg/project
Inspect: https://vercel.com/myorg/project/2x6dq3wmp [3s]
- Building
- Building
- Completing
Preview: https://my-project-domain.now.sh [6s]
To deploy to production (my-project-domain.now.sh), run `now --prod`
https://project-2y6dq3wmp.now.sh
```

So this works fine for now, although I probably like [@michaellopez 's solution](https://github.com/vercel/vercel/discussions/4523#discussioncomment-20480) better.

**Comment by michaellopez** [comment, marked as answer]
@borekb @coetry Now has always had support for getting the deployment URL when run in a subshell.

`DEPLOYMENT_URL=$(now ...)`

or nowadays...

`DEPLOYMENT_URL=$(vercel ...)`

That's what we've been doing for years and it works great and is reliable. Can't find the source for it though, I've looked here and there over the years without success.

**Reply by borekb** [reply]
That's a great tip, will try it out!

**Reply by mattdell** [reply]
I can confirm this works for me as well. Great!
